src/compiler/checker.ts

```diff
@@ -16023,7 +16018,8 @@ namespace ts {
                 const constraint = getConstraintOfTypeParameter(inference.typeParameter);
                 if (constraint) {
                     const instantiatedConstraint = instantiateType(constraint, context.nonFixingMapper);
-                    if (!inferredType || !context.compareTypes(inferredType, getTypeWithThisArgument(instantiatedConstraint, inferredType))) {
+                    const isCompletionContext = contextFlags && ((contextFlags | ContextFlags.Completion) === contextFlags);

```

Comment:
Why not just `contextFlags & ContextFlags.Completion`?

Reply:
That's very true! It still has to be `contextFlags && (contextFlags & ContextFlags.Completion)` as `contextFlags` can be undefined.
Just updated it.